src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts

```diff
@@ -149,6 +150,47 @@ describe('ConfidentialTransaction class', () => {
     });
   });
 
+  describe('method: getInvolvedParties', () => {
+    it('should get involved parties for the transaction', async () => {
+      const rawId = dsMockUtils.createMockU64(transaction.id);
+      dsMockUtils.createQueryMock('confidentialAsset', 'transactionParties', {
+        entries: [
+          tuple(
+            [rawId, dsMockUtils.createMockIdentityId('0x01')],
+            dsMockUtils.createMockBool(true)
+          ),
+          tuple(
+            [rawId, dsMockUtils.createMockIdentityId('0x02')],
+            dsMockUtils.createMockBool(true)
+          ),
+        ],
+      });
+
+      const result = await transaction.getInvolvedParties();
+
+      expect(result).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({ did: '0x01' }),
+          expect.objectContaining({ did: '0x02' }),
+        ])
+      );
+    });
+
+    it('should throw an error if no parties are found', async () => {
+      dsMockUtils.createQueryMock('confidentialAsset', 'transactionParties', {
+        entries: [],
+      });
+
+      const expectedError = new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message:
+          'No involved parties were found for this transaction. Its likely been completed and the chain storage has been pruned',
+      });
```

Comment:
I thought this should fail as we didn't add the `data` part here

Reply:
data doesn't get compared in the error comparision, just code and message.